### PR TITLE
Feature issue #29 deploy webpack aws s3

### PR DESCRIPTION
**Pull request by wintondeshong**
Addition of deploy to aws-s3 and file/directory copying commands for issue #29

### CODE REVIEW

File: cli-copy.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env node
+
+/**************************************************************************************************
+ * Imports
+ **************************************************************************************************/
+
+const commands      = require("./_modules/commands");
+const echo          = require("./_modules/echo");
+const program       = require("commander");
+const shell         = require("shelljs");
```
Comment (brandongregoryscott) [inline cli-copy.js]: Minor, but there's extra spacing here

File: _modules/commands.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 module.exports = {
+    copy: {
+        command:     "copy",
+        description: "Copy files and/ or directories",
```
Comment (Stefanie899) [inline _modules/commands.js]: Super minor, but isn't this usually written 'and/or'?

File: cli-copy.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+/**************************************************************************************************
+ * Commands
+ **************************************************************************************************/
+
+// #region NuGet commands
```
Comment (brandongregoryscott) [inline cli-copy.js]: Looks like this was a copy/paste typo

File: _modules/webpack-publish.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    cmd() {
+        return "npm run build";
+    },
+    description() {
+        return `Publishes a release build of the frontend project (via ${this.cmd()}) in ${frontendPath.projectDir()}`;
+    },
+    run() {
+        // Clean publish directory
+        echo.message(`Cleaning publish directory ${frontendPath.publishDir()}...`);
+        shell.rm("-rf", frontendPath.publishDir());
```
Comment (brandongregoryscott) [inline _modules/webpack-publish.js]: Should we be checking the exit status of this `rm` call before continuing?

File: cli-copy.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+const copy = {
+    description() {
+        return "Copies files and/ or directories";
+    },
+    run(source, destination, options) {
```
Comment (brandongregoryscott) [inline cli-copy.js]: Consider adding a JSDoc comment regarding what `source`, `destination`, and `options` are

File: cli-deploy-aws-s3.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+        return true;
+    },
+};
+
+// #endregion Commands
+
+
+/**************************************************************************************************
+ * Entrypoint / Command router
```
Comment (Stefanie899) [inline cli-deploy-aws-s3.js]: Do we need this comment as well as the #region? It kinda feels like they both serve the same purpose, to differentiate the block of code, so it feels redundant to me.

Comment (wintondeshong) [reply cli-deploy-aws-s3.js]: Its a compromise honestly. I don't personally leverage #regions like this and find code more readable with the block comments.

**Pull request merged**